plugins/remark-lint-validate-internal-links/index.js

```diff
@@ -0,0 +1,273 @@
+import { accessSync, constants, readFileSync } from 'fs';
+import { join, extname } from 'path';
+import { lintRule } from 'unified-lint-rule';
+import { visit } from 'unist-util-visit';
+import { fromMarkdown } from 'mdast-util-from-markdown';
+import warnings from './warnings.js';
+import debugModule from 'debug';
+import GithubSlugger from 'github-slugger';
+
+const log = new debugModule('remark-lint:validate-internal-links');
+const slugger = new GithubSlugger();
+let availableAnchors = []
```

Comment:
**issue**: unless there's some magic being done by remark-lint, this file is only required once.  this is building a list of anchors globally.  it should be anchors per-file.  you _might_ be able to work it into your per-file cache system, but that might be a premature optimization.  I'd be happy if it was scoped in `validateInternalLinks`.